New Feature - quick sharing bookmark from bookmarks screen..

pastmaster007 opened the pull request:
<!--
Note: This checklist is a reminder of our shared engineering expectations. Feel free to change it, although assigning a GitHub reviewer and the items in bold are required.
-->
**Description**:
Ability to share a bookmark on swipe from left to right in bookmarks screen.
This feature supports only in iOS 11.0 and above.
Added corresponding new pixel identifier called "bm_sh" (bookmark share)

**Steps to test this PR**:
1. On iOS 11 and above supported device, open bookmarks screen by pressing bookmarks icon on home screen tab bar
1. Now swipe left to right on a bookmark results share sheet appears.

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

Commit:
New Feature
Ability to share a bookmark on swipe from left to right in bookmarks screen.
This feature supports only in iOS 11.0 and above.
Added corresponding new pixel identifier called "bm_sh" (bookmark share)

bwaresiak commented on Core/Pixel.swift:
```diff
@@ -88,6 +88,7 @@ public enum PixelName: String {
     case homeScreenDeleteFavorite = "mh_df"
     case homeScreenFavoriteMoved = "mh_fm"
     
+    case bookmarkShare = "bm_sh"
```
You don't have to worry about pixel requests - these are needed only when we want to check feature usage after a release. It's OK to remove it from this and other PRs.

bwaresiak commented on DuckDuckGo/BookmarksViewController.swift:
```diff
@@ -50,6 +50,17 @@ class BookmarksViewController: UITableViewController {
+            self.showShareSheet(for: indexPath)
+            completionHandler(true)
+        }
+        shareContextualAction.backgroundColor = UIColor.ppGreen
```
Please use 'cornflowerBlue' instead.

bwaresiak commented on DuckDuckGo/BookmarksViewController.swift:
```diff
@@ -50,6 +50,17 @@ class BookmarksViewController: UITableViewController {
+    
+    @available(iOS 11.0, *)
+    override func tableView(_ tableView: UITableView, leadingSwipeActionsConfigurationForRowAt indexPath: IndexPath) -> UISwipeActionsConfiguration? {
+        let shareContextualAction = UIContextualAction.init(style: .normal, title: "Share") { (_, _, completionHandler) in
```
Please drop the `.init`

bwaresiak commented on DuckDuckGo/BookmarksViewController.swift:
```diff
@@ -50,6 +50,17 @@ class BookmarksViewController: UITableViewController {
+            completionHandler(true)
+        }
+        shareContextualAction.backgroundColor = UIColor.ppGreen
+        return UISwipeActionsConfiguration.init(actions: [shareContextualAction])
```
Please drop the `.init`

bwaresiak commented on DuckDuckGo/BookmarksViewController.swift:
```diff
@@ -50,6 +50,17 @@ class BookmarksViewController: UITableViewController {
+    @available(iOS 11.0, *)
+    override func tableView(_ tableView: UITableView, leadingSwipeActionsConfigurationForRowAt indexPath: IndexPath) -> UISwipeActionsConfiguration? {
+        let shareContextualAction = UIContextualAction.init(style: .normal, title: "Share") { (_, _, completionHandler) in
+            Logger.log(text: "share action")
```
I believe we can remove that and only log error cases, rather than regular usage.

bwaresiak reviewed (changes requested):
Thank you for making this PR. I've left some comments inline.

Commit:
removed .init on share action and swipe action
changed swipe to share action background color to match design guidelines..
removed pixel tracking for this feature..
now app reads share text on swipe from multi-lingual file

bwaresiak reviewed (approved):
This looks good!

Sorry it took me a while to get back to you. Thanks for your contribution!
